src/sql/src/plan/statement/ddl.rs

```diff
@@ -1018,6 +1018,16 @@ pub fn plan_create_source(
     }))
 }
 
+fn ensure_ccsr_options(ccsr_with_options: &BTreeMap<String, Value>) -> Result<(), anyhow::Error> {
```

Comment:
I think it might make sense to move this function next to the `with_options!` macro in `sql::normalize` and rename it to `ensure_empty_options(options: &Btreemap<String, Value>, for: &str)` and then call it like `ensure_empty_options(&ccsr_options, "CONFLUENT SCHEMA REGISTRY")?;`.